### ISSUE TITLE
Add an InvokeIf method to FluentExtensions

### ISSUE THREAD

**Issue by rdeago** [Tenacom/Louis#52]
### Reference version

1.0.93-preview

### Background and motivation

We have encountered a number of situations where we want to call `FluentExtensions.Invoke` only if a certain condition is satisfied.

Currently, this can be achieved either by adding an `if` or a ternary operator inside the lambda passed to `Invoke` (ugly!), or via the combined use of `FluentExtensions.If` and `FluentExtensions.Invoke` (a bit leaner, but far from optimal).

### Proposed enhancement

Ideally we need an extension method with the following signature:

```c#
public static T InvokeIf<T>(this T @this, bool condition, Action<T> action);
```

Furthermore, additional parameters may need to be passed to `action`:

```c#
public static T InvokeIf<T, TArg>(this T @this, bool condition, TArg arg, Action<T, TArg> action);
public static T InvokeIf<T, TArg1, TArg2>(this T @this, bool condition, TArg1 arg1, TArg2 arg2, Action<T, TArg1, TArg2> action);
// ... and so on (currently a maximum of 3 additional parameters is used in existing code)
```

### Implementation proposals

Implementation should be pretty straightforward, especially after (or even during) the implementation of #51.

### Usage examples

```c#
new StringBuilder()
    .Append("Your account has been ")
    .InvokeIf(amount < 0, amount, (x, n) => x.Append("charged ").AppendFormat(CultureInfo.CurrentCulture, "C", Math.Abs(n))
    .InvokeIf(amount > 0, amount, (x, n) => x.Append("credited ").AppendFormat(CultureInfo.CurrentCulture, "C", n)
    .InvokeIf(amount == 0, x => x.Append("left untouched"));
```

### Risks

No risks that I am aware of.

**Comment by rdeago** [comment]
@ric15ni this can be closed in the same PR as #51.

**Comment by ric15ni** [comment]
This could be closed in the same PR as #51.
